### ISSUE TITLE
 QTv0.6alpha 1aed542 does not seem to remove pink highlights

### ISSUE THREAD

**Issue by iFloris** [ilia3101/MLV-App#2]
Hello! First off, huge huge thank you for building these apps @ilia3101 @bouncyball-git and @masc4ii . 💯  This is sorely needed. I originally thought that footage.app would be **the** app, but with the whole licensing thing and the developer choosing to go a different direction (which is of course absolutely his right and to be respected), I was afraid that the idea of easy-to-use-mlv-to-something-fcpx-can-use was going out the window.
In comes MLV-app. ❤️ 

Anyway, I tried the cocoa app on my mac pro 3,1, which looks awesome. But! It currently does not seem to support batch exporting, which is something that I really need after a day of filming. The QT app does support batch quite well. 

I corrected my mlv images (5d3 8-10 bit 3,5k on 24-70 2.8) and set MLV-app qt to export some 40 MLV's to mov. Awesome! 👏 

However, after having it run for the night and inspecting the exported mov files, I noticed that the pink highlights are still very much noticeable. I checked the settings, remove highlights was indeed checked. [Example, see car to the right](https://www.dropbox.com/s/ugjg9ebo6qvncix/Schermafdruk%202017-08-31%2018.17.23.jpg?dl=0)

So I think that might be a bug in MLV-App QTv0.6a.

TL:DR
Exported 40 MLV to mov using QTv0.6a. Pink highlights still present. User error or bug?

**Comment by masc4ii** [comment]
Hej,
thanks for the nice feedback! 
Yes, in your picture that looks like highlight reconstruction does not work. Can you minimize this mlv file to 5 pics using mlv_dump and send it to us? I only have 5D2 files and with that I can't reproduze this behaviour. Can you write which steps you do until you export? This could help too. Thanks!

**Comment by iFloris** [comment]
Hello @masc4ii thanks for your reply. Could you specify the command required for mlv_dump? I have never used mlv_dump cli before. I tried `mlv_dump.macos -f 5 path/to/file.mlv` but that gave me `Reached end of chunk 1/1 after 1076 blocks
Processed 0 video frames at 24.01 FPS (0.00 s)
Done` 
So I think that was not what was expected. Apologies if this seems backward 😄 
I was exporting some more files using MLV-app, so I took a [screenshot](https://www.dropbox.com/s/iz9mswgs2t16ltz/Schermafdruk%202017-08-31%2022.58.27.jpg?dl=0) in which you can see the settings I used. The steps I took were:

1. Import MLV files
2. double click each MLV and correct exposure and check 'Highlight reconstruction'
3. Export to mov

**Comment by masc4ii** [comment]
Use this: `mlv_dump -f 5 -o OUTPUT.MLV INPUT.MLV`
Whooohoouu... this looks really pinkish. I have never seen something like  that!

**Comment by iFloris** [comment]
Thanks, I guess I missed the -o option 😇 
Yes, that screenshot is from one of my first tests with filming mlv in 3.5k, I basically had no idea how to expose raw video properly yet. It is kind of hard to see with the hacked preview, but I have grown more comfortable with it by trial and error, and by trusting the raw histogram more.
I have uploaded a couple 5 frame clips [here](https://www.dropbox.com/s/zoo6peq4eq9yw26/mlv-masc4ii.zip?dl=0), I hope these are useful.

**Comment by masc4ii** [comment]
Thank you. But... are you able to open these files? I only get noise and the metadata (using the latest revision, not 0.6, maybe something is broken). :-(

**Comment by iFloris** [comment]
Oh I have not tried! 😲 I just set it to export as per your instructions 😆 

MLV-App does not seem to be able to open them, [as you can see here](https://www.dropbox.com/s/s94hdfm1qub14b0/Schermafdruk%202017-09-01%2019.34.58.jpg?dl=0), I get pink noise as well. 

[But Footage.app can](https://www.dropbox.com/s/khkltu857trge4c/Schermafdruk%202017-09-01%2019.34.05.jpg?dl=0).
I will put the smallest of the original MLV files from my first test (which have loads of pink highlights) in a dropbox folder for you, but this will take a while as I have terrible upload speeds.

[You can find the dropbox folder here](https://www.dropbox.com/sh/cjzvvtbps917h35/AABkpOMUbCh6ydyjFUZrIF5Ra?dl=0).
I will update this comment when uploading has completed.

**Comment by ilia3101** [comment]
I have tested in Cocoa GUI, and i could not open mlv_dump files either, which is strange as I am sure i've done that before.

Anyway, I think this should be solved if a new version would be compiled. **[Thanks to this commit.](https://github.com/ilia3101/MLV-App/commit/6f2535c55530554fad426231dc67ddb5e2ea5211#diff-f8554ba11b522aafab1439ca7736b256)**

Tested in Cocoa GUI:
![picture of mlv app](https://image.ibb.co/kNLCPF/Screen_Shot_2017_09_03_at_16_37_12.png)

**Comment by masc4ii** [comment]
Okay, with the huuuuuge file direct from the camera it seems to work also with the latest Qt revision.
![Picture](https://image.ibb.co/f58MZF/Bildschirmfoto_2017_09_03_um_20_00_27.png)
So I think this issue can be closed.

**Comment by iFloris** [comment]
Looks awesome! I will give it a try tomorrow afternoon, but I was never able to get it looking that clean as you both did. Should I use a freshly built from source version? Thanks a lot for the efforts so far @masc4ii and @ilia3101 👍

**Comment by masc4ii** [comment]
For now, you can just check out the latest revision and compile it yourself, or you must wait for the next release. But there are some other features in work, so the next release will need some time...

**Comment by ilia3101** [comment]
Ok I'm closing the issue now. @iFloris Are you able to compile Qt version yourself?

**Comment by iFloris** [comment]
😞  I thought I knew how to build, but I can only follow the _easy mode_ instructions in readme.md. I have no clue how to build the qt version. But I tried a freshly pulled and built cocoa version! 

It.. crashes as soon as I open a MLV file.. sorry!

Here is the crash report. I tried this on my macbook air using the large MLV (M30-1022) you used above.

`Process:               MLV App [31708]
Path:                  /Users/USER/*/MLV App.app/Contents/MacOS/MLV App
Identifier:            fm.ilia.MLV App
Version:               [Sep  4 2017 @metacortex-9.local]
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           MLV App [31708]
User ID:               501

Date/Time:             2017-09-04 20:30:26.590 +0200
OS Version:            Mac OS X 10.12.6 (16G29)
Report Version:        12
Anonymous UUID:       
Sleep/Wake UUID:       
Time Awake Since Boot: 300000 seconds
Time Since Wake:       5900 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000068
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [0]

VM Regions Near 0x68:
--> 
    __TEXT                 000000010f873000-000000010f888000 [   84K] r-x/rwx SM=COW  /Users/USER/*/MLV App.app/Contents/MacOS/MLV App

Application Specific Information:
Performing @selector(ok:) from sender NSButton 0x608000155be0

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   libsystem_c.dylib             	0x00007fffb7ce66a0 flockfile + 4
1   libsystem_c.dylib             	0x00007fffb7ce983c fwrite + 72
2   fm.ilia.MLV App               	0x000000010f877011 writeMlvAudioToWave + 497
3   fm.ilia.MLV App               	0x000000010f881f23 __38-[NSButton(mainMethods) openMlvDialog]_block_invoke + 403
4   com.apple.AppKit              	0x00007fffa06f0a9c -[NSSavePanel _dismissModeless:] + 124
5   com.apple.AppKit              	0x00007fffa06f2ff4 -[NSSavePanel ok:] + 334
6   libsystem_trace.dylib         	0x00007fffb7ea63a7 _os_activity_initiate_impl + 53
7   com.apple.AppKit              	0x00007fffa076b721 -[NSApplication(NSResponder) sendAction:to:from:] + 456
8   com.apple.AppKit              	0x00007fffa024fcc4 -[NSControl sendAction:to:] + 86
9   com.apple.AppKit              	0x00007fffa024fbec __26-[NSCell _sendActionFrom:]_block_invoke + 136
10  libsystem_trace.dylib         	0x00007fffb7ea63a7 _os_activity_initiate_impl + 53
11  com.apple.AppKit              	0x00007fffa024fb44 -[NSCell _sendActionFrom:] + 128
12  com.apple.AppKit              	0x00007fffa0292539 -[NSButtonCell _sendActionFrom:] + 98
13  libsystem_trace.dylib         	0x00007fffb7ea63a7 _os_activity_initiate_impl + 53
14  com.apple.AppKit              	0x00007fffa024e426 -[NSCell trackMouse:inRect:ofView:untilMouseUp:] + 2481
15  com.apple.AppKit              	0x00007fffa0292272 -[NSButtonCell trackMouse:inRect:ofView:untilMouseUp:] + 798
16  com.apple.AppKit              	0x00007fffa024cddb -[NSControl mouseDown:] + 832
17  com.apple.AppKit              	0x00007fffa08e724f -[NSWindow(NSEventRouting) _handleMouseDownEvent:isDelayedEvent:] + 6341
18  com.apple.AppKit              	0x00007fffa08e3a6c -[NSWindow(NSEventRouting) _reallySendEvent:isDelayedEvent:] + 1942
19  com.apple.AppKit              	0x00007fffa08e2f0a -[NSWindow(NSEventRouting) sendEvent:] + 541
20  com.apple.AppKit              	0x00007fffa0767681 -[NSApplication(NSEvent) sendEvent:] + 1145
21  com.apple.AppKit              	0x00007fff9ffe2427 -[NSApplication run] + 1002
22  fm.ilia.MLV App               	0x000000010f876e05 NSApplicationMain + 10853
23  fm.ilia.MLV App               	0x000000010f874399 main + 9
24  libdyld.dylib                 	0x00007fffb7c74235 start + 1`

**Comment by ilia3101** [comment]
@iFloris Sorry about that, I know what is causing it, it was just  a test I did, but here's how to fix it: delete line 119 in `platform/cocoa/main_methods.m`
It was just a test for initial audio support, I'll get rid of it in next commit.

As for Qt version, there is a tutorial for linux https://seescho.wordpress.com/fotos/ml/ 
It should be 100% the same to do it on Mac, however for Qt version you need to install Qt creator

**Comment by iFloris** [comment]
Yes, got it working! Thanks! Also, no more pink highlights! Awesome! 👏 
I will check out QT creator in order to batch export the MLV files I shot today and see if I can get it to work, otherwise I will just hand pick the best shots and export them using cocoa. Again, thanks a lot!
